// @adonisjs/mail/build/src/Drivers/Fake.js
"use strict";
/*
 * @adonisjs/mail
 *
 * (c) Harminder Virk <virk@adonisjs.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FakeDriver = void 0;
/// <reference path="../../adonis-typings/mail.ts" />
const nodemailer_1 = __importDefault(require("nodemailer"));
/**
 * Smtp driver to send email using smtp
 */
class FakeDriver {
    constructor(listener) {
        this.listener = listener;
        this.transporter = nodemailer_1.default.createTransport({
            jsonTransport: true,
        });
    }
    /**
     * Send message
     */
    async send(message) {
        if (!this.transporter) {
            throw new Error('Driver transport has been closed and cannot be used for sending emails');
        }
        const listenerResponse = this.listener(message);
        const response = await this.transporter.sendMail(message);
        return { ...response, ...listenerResponse };
    }
    /**
     * Close transporter connection, helpful when using connections pool
     */
    async close() {
        this.transporter.close();
        this.transporter = null;
    }
}
exports.FakeDriver = FakeDriver;
